CMakeLists.txt

```diff
@@ -47,7 +47,7 @@ if(USE_GPU)
 endif(USE_GPU)
 
 if(UNIX OR MINGW OR CYGWIN)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -O3 -Wall -std=c++11 -Wno-ignored-attributes")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -O3 -Wall -std=c++11 -Wno-ignored-attributes -march=native")
```

Comment:
@guolinke Perhaps `-mtune=native` is better for those using portable installations. We could do this:

* `-march=core2` so anyone benefits from core2 instructions
* `-mtune=native` so it is tailored for the current CPU used during compilation, and fallbacks to core2 if native instructions are not available